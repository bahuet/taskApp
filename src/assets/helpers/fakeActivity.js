export default ({ todos }) => {


  return (
    null
  )
}