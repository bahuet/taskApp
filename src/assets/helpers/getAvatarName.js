import React, { useState } from 'react'

// assigner un avatar aléatoire, code éphémère pour la démo.

export default () => {
  const avatarList = []
  for (let i = 0; i <= 20; i++) {
    avatarList.push(`${i}.png`)
  }
  return avatarList[Math.floor(Math.random() * avatarList.length)]
}

