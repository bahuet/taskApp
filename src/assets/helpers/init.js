export default (namesNum, tasksNum) => {

  // Il n'est pas impossible que j'aie légérement overengineered cette partie

  // generate initial values for demo purposes
  const randomValueFromArray = array => array[Math.floor(Math.random() * array.length)]

  const createRandomName = () => {
    const firstNames = ["Adam", "Alex", "Aaron", "Ben", "Carl", "Dan", "David", "Edward", "Fred", "Frank", "George", "Hal", "Hank", "Ike", "John", "Jack", "Joe", "Larry", "Monte", "Matthew", "Mark", "Nathan", "Otto", "Paul", "Peter", "Roger", "Roger", "Steve", "Thomas", "Tim", "Ty", "Victor", "Walter"]
    const lastNames = ["Anderson", "Ashwoon", "Aikin", "Bateman", "Bongard", "Bowers", "Boyd", "Cannon", "Cast", "Deitz", "Dewalt", "Ebner", "Frick", "Hancock", "Haworth", "Hesch", "Hoffman", "Kassing", "Knutson", "Lawless", "Lawicki", "Mccord", "McCormack", "Miller", "Myers", "Nugent", "Ortiz", "Orwig", "Ory", "Paiser", "Pak", "Pettigrew", "Quinn", "Quizoz", "Ramachandran", "Resnick", "Sagar", "Schickowski", "Schiebel", "Sellon", "Severson", "Shaffer", "Solberg", "Soloman", "Sonderling", "Soukup", "Soulis", "Stahl", "Sweeney", "Tandy", "Trebil", "Trusela", "Trussel", "Turco", "Uddin", "Uflan", "Ulrich", "Upson", "Vader", "Vail", "Valente", "Van Zandt", "Vanderpoel", "Ventotla", "Vogal", "Wagle", "Wagner", "Wakefield", "Weinstein", "Weiss", "Woo", "Yang", "Yates", "Yocum", "Zeaser", "Zeller", "Ziegler", "Bauer", "Baxster", "Casal", "Cataldi", "Caswell", "Celedon", "Chambers", "Chapman", "Christensen", "Darnell", "Davidson", "Davis", "DeLorenzo", "Dinkins", "Doran", "Dugelman", "Dugan", "Duffman", "Eastman", "Ferro", "Ferry", "Fletcher", "Fietzer", "Hylan", "Hydinger", "Illingsworth", "Ingram", "Irwin", "Jagtap", "Jenson", "Johnson", "Johnsen", "Jones", "Jurgenson", "Kalleg", "Kaskel", "Keller", "Leisinger", "LePage", "Lewis", "Linde", "Lulloff", "Maki", "Martin", "McGinnis", "Mills", "Moody", "Moore", "Napier", "Nelson", "Norquist", "Nuttle", "Olson", "Ostrander", "Reamer", "Reardon", "Reyes", "Rice", "Ripka", "Roberts", "Rogers", "Root", "Sandstrom", "Sawyer", "Schlicht", "Schmitt", "Schwager", "Schutz", "Schuster", "Tapia", "Thompson", "Tiernan", "Tisler"]
    return randomValueFromArray(firstNames) + ' ' + randomValueFromArray(lastNames)
  }

  let userId = 1100000
  let taskId = 2100000

  const getNewUserId = () => userId++
  const getNewTaskId = () => taskId++


  const createTask = (taskText) => {
    //Une tâche ne peut pas être 'completed' sans être 'focused', on le prend en compte pendant la génération des valeurs initiales
    const fixedRandomValue = Math.floor(Math.random() * 3)
    const randomUser = randomValueFromArray(fakeUserssArray)
    return (
      {
        id: getNewTaskId(),
        userName: randomUser.name,
        userId: randomUser.id,
        text: taskText,
        completed: fixedRandomValue ? false : (Math.floor(Math.random() * 2) ? false : true),
        urgent: Math.floor(Math.random() * 3) ? false : true,
        focus: fixedRandomValue ? false : true
      })
  }

  const createUser = (name) => {
    const roles = ['Manager', 'Designer', 'Comptable', 'CEO', 'CSO', 'DRH', 'R&D', 'Développeur front', 'Développeur front']
    return (
      {
        id: getNewUserId(),
        name: name || createRandomName(),
        role: randomValueFromArray(roles)
      })
  }

  const createRandomTask = () => createTask(`Tâche aléatoire ${taskId} générée automatiquement pour la démo`)

  const fakeDataGen = (initialArray, randomValueFn, number) => {
    const out = [...initialArray]
    for (let i = 0; i < number; i++) {
      const value = randomValueFn()
      out.push(value)
    }
    return out
  }

  const initialUserNames = ['Eloi', 'George Foreman', 'Dmitri Mendeleev', '吕小军',]
    .map(name => createUser( name))
  const fakeUserssArray = fakeDataGen(initialUserNames, createUser, namesNum)

  let initialTasks = ['Relancer le fournisseur pour savoir les dates de livraison', 'Régler la facture du fournisseur', 'Refaire du café', "Traduire la notice d'utilisation en anglais"]
    .map(task => createTask(task))

  const fakeTasksArray = fakeDataGen(initialTasks, createRandomTask, tasksNum)

  return [fakeUserssArray, fakeTasksArray]
}