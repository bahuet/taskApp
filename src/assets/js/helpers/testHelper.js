export const user = { name: "userNameTest", id: 1, role: "userRoleTest", avatar: "1.png" }
export const userTodos = [
  {
    id: 10,
    userId: 20,
    userName: "fakeName1",
    text: "testText",
    completed: false,
    urgent: false,
    focus: false
  },
  {
    id: 11,
    userId: 21,
    userName: "fakeName2",
    text: "testText",
    completed: false,
    urgent: false,
    focus: false
  }
]
