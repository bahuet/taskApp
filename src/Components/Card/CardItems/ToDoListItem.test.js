import React from "react"
import "@testing-library/jest-dom/extend-expect"
import { render, fireEvent } from "@testing-library/react"
import TodoListItem from "./ToDoListItem"

import { user, userTodos } from "../../../assets/js/helpers/testHelper"
const todo = userTodos[0]
describe("<ToDoListItem />", () => {
  test("renders user task values", () => {
    const component = render(<TodoListItem user={user} todo={todo} admin={true} />)
    expect(component.container).toHaveTextContent(todo.text)
  })

  test("renders admin options", () => {
    const component = render(<TodoListItem user={user} todo={todo} admin={true} />)
    expect(component.container).toHaveTextContent(todo.text)
  })

  test("Icons not visible by default", () => {
    const { getByTestId } = render(<TodoListItem user={user} todo={todo} admin={true} />)
    const element = getByTestId("icons-container")
    expect(element).not.toBeVisible()
  })

  test("Icons visible on mouseover", () => {
    const { getByTestId } = render(<TodoListItem user={user} todo={todo} admin={true} />)
    const element = getByTestId("icons-container")
    fireEvent.mouseOver(element)
    expect(element).toBeVisible()
  })
})
