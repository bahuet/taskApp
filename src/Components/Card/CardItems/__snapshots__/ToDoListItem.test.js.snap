// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ToDoListItem /> renders admin icons on mouseover 1`] = `
<li
  className="MuiListItem-root makeStyles-listItem-221 MuiListItem-dense MuiListItem-gutters MuiListItem-divider"
  disabled={false}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  <div
    className="MuiListItemText-root makeStyles-multiline-220 MuiListItemText-dense"
  >
    <span
      className="MuiTypography-root MuiListItemText-primary MuiTypography-body2"
    >
      testText
    </span>
  </div>
  <div
    style={
      Object {
        "opacity": 0,
        "visibility": "hidden",
      }
    }
  >
    <div
      style={
        Object {
          "display": "flex",
          "flexDirection": "column",
        }
      }
    >
      <button
        aria-describedby={null}
        className="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeSmall MuiIconButton-edgeStart"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseOver={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        title="Marquer \\"Urgent\\""
        type="button"
      >
        <span
          className="MuiIconButton-label"
        >
          <svg
            aria-hidden="true"
            className="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall"
            focusable="false"
            role="presentation"
            viewBox="0 0 24 24"
          >
            <path
              d="M1 21h22L12 2 1 21zm12-3h-2v-2h2v2zm0-4h-2v-4h2v4z"
            />
          </svg>
        </span>
        <span
          className="MuiTouchRipple-root"
        />
      </button>
      <button
        aria-describedby={null}
        className="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeSmall MuiIconButton-edgeStart"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseOver={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        title="TransfÃ©rer"
        type="button"
      >
        <span
          className="MuiIconButton-label"
        >
          <svg
            aria-hidden="true"
            className="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall"
            focusable="false"
            role="presentation"
            viewBox="0 0 24 24"
          >
            <path
              d="M15.5 5H11l5 7-5 7h4.5l5-7z"
            />
            <path
              d="M8.5 5H4l5 7-5 7h4.5l5-7z"
            />
          </svg>
        </span>
        <span
          className="MuiTouchRipple-root"
        />
      </button>
      <button
        aria-describedby={null}
        className="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeSmall MuiIconButton-edgeStart"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseOver={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        title="Supprimer ðŸ”’"
        type="button"
      >
        <span
          className="MuiIconButton-label"
        >
          <svg
            aria-hidden="true"
            className="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall"
            focusable="false"
            role="presentation"
            viewBox="0 0 24 24"
          >
            <path
              d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zM19 4h-3.5l-1-1h-5l-1 1H5v2h14V4z"
            />
          </svg>
        </span>
        <span
          className="MuiTouchRipple-root"
        />
      </button>
    </div>
  </div>
</li>
`;
